rest_api/controller/documents.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+from haystack.schema import Document
+from haystack.document_store import sql
+from rest_api.config import LOG_LEVEL
+from typing import Dict
+from fastapi import APIRouter, HTTPException
+import logging
+
+sql = sql.SQLDocumentStore()
+now = datetime.now()
+SUCCESS: int = 200
+CREATED: int = 201
+
+logger = logging.getLogger('haystack')
+logger.setLevel(LOG_LEVEL)
+
+router = APIRouter()
+
+
+@router.post("/documents", status_code=CREATED)
+def create_document(payload: Dict[Document]):
+    """
+    Creates a Document object
+    :param payload: input from the client to create a document.
+    """
+    logger.info("Constructing document object")
+    created_document = Document.from_dict(payload)
+    logger.info("Constructed!")
+    return {"created": created_document,
+            "created_at": now}
+
+
+@router.get("/documents", status_code=SUCCESS)
+def get_all_documents(start: int = 0, limit: int = 20):
+    """
+    Returns the documents which are available in the database
+    Supports pagination
+    :param start: used to set the starting point
+    :param limit: used to delimit the number of retrieved documents
+    """
+    try:
+        logger.info("Retrieving documents")
+        retrieved_documents = sql.get_all_documents()
```

Comment:
We can build pagination on to of generator. https://github.com/deepset-ai/haystack/blob/6d00eff79602d00251c09ce1071aa7f44e18f158/haystack/document_store/elasticsearch.py#L508